### PR TITLE
Fix Request-URI Too long for batch edit and merge object

### LINKED ISSUE #5500: 414 Request-URI Too Long
When mass editing attributes, MISP does a GET request like:
```
https://misp.local/attributes/editSelected/28854/[2113775,2113833,2113834,2113835,2113830,2113831,2113832,2113827,2113828,2113829,2113824,2113825,2113826,2113821,2113822,2113823,2113818,2113819,2113820,2113815,2113816,2113817,2113812,2113813,2113814,2113791,2113792,2113793,2113809,2113810,2113811,2113806,2113807,2113808,2113803,2113804,2113805,2113800,2113801,2113802,2113797,2113798,2113799,2113794,2113795,2113796,2113785,2113786,2113787,2113788,2113789,2113790,2113782,2113783,2113784,2113779,2113780,2113781,2113776,2113777,2113778,2113774,2113773,2113772,2113764,2113771,2113770,2113769,2113768,2113767,2113766,2113765,2113763]?_=1578668575760
```
This can get quite long. Apache has a default [LimitRequestLine ](http://httpd.apache.org/docs/2.4/mod/core.html#limitrequestline)of 8190 bytes, but [CIS Apache Benchmark 1.10.1](https://www.cisecurity.org/benchmark/apache_http_server/) limits this to 512 bytes.

Perhaps MISP can use of POST for this?

### PR DESCRIPTION
**Pull request by mokaddem**
- Allow POST on `getMassEditForm`
- Usage of session to keep track of data to be merged
    - As we can't POST data to be merged to `objects/edit` with the current object

Fix #5500

### CODE REVIEW

File: app/Controller/AttributesController.php [hunk 1]
```diff
@@ -1399,11 +1400,141 @@ public function deleteSelected($id = false, $hard = false)
...
         if (!$this->request->is('ajax')) {
             throw new MethodNotAllowedException(__('This method can only be accessed via AJAX.'));
         }
+        if (!$this->request->is('post')) {
+            throw new MethodNotAllowedException();
+        }
+        if (!isset($id)) {
+            throw new MethodNotAllowedException(__('No event ID provided.'));
+        }
+        $event = $this->Attribute->Event->fetchSimpleEvent($this->Auth->user(), $id, $params = array(
```
Comment (JakubOnderka) [inline app/Controller/AttributesController.php]: Check if event exists is missing here.

File: app/Controller/AttributesController.php [hunk 2]
```diff
@@ -1399,11 +1400,141 @@ public function deleteSelected($id = false, $hard = false)
...
+            );
+            if ($cluster['synonyms_string'] !== '') {
+                $temp['infoContextual'] = __('Synonyms: ') . $cluster['synonyms_string'];
+            }
+            $clusterItemsRemove[] = $temp;
+        }
+        unset($clusters);
+        $conditions = array();
+        if (!$this->_isSiteAdmin()) {
+            $conditions = array('Tag.org_id' => array(0, $this->Auth->user('org_id')));
```
Comment (JakubOnderka) [inline app/Controller/AttributesController.php]: It should be `$conditions[]`

File: app/Controller/AttributesController.php [hunk 3]
```diff
@@ -1399,11 +1400,141 @@ public function deleteSelected($id = false, $hard = false)
...
+                        )
+                    ),
+                )
+
+            );
+        }
+
+        // clusters to add
+        $this->GalaxyCluster = ClassRegistry::init('GalaxyCluster');
+        $clusters = $this->GalaxyCluster->find('all', array(
```
Comment (JakubOnderka) [inline app/Controller/AttributesController.php]: What about to use `list`?

File: app/Controller/ObjectsController.php [hunk 4]
```diff
@@ -397,15 +404,25 @@ public function edit($id, $update_template_available=false, $onlyAddNewAttribute
         $templateData = $this->MispObject->resolveUpdatedTemplate($template, $object, $update_template_available);
         $this->set('updateable_attribute', $templateData['updateable_attribute']);
         $this->set('not_updateable_attribute', $templateData['not_updateable_attribute']);
-        if (isset($this->params['named']['revised_object'])) {
-            $revisedData = $this->MispObject->reviseObject($this->params['named']['revised_object'], $object, $template);
+        if (!empty($this->Cookie->read('object_being_created')) && !empty($this->params['named']['cur_object_tmp_uuid'])) {
```
Comment (JakubOnderka) [inline app/Controller/ObjectsController.php]: I think it should be better to save that data to session, not to cookie.

Comment (mokaddem) [reply app/Controller/ObjectsController.php]: Totally!

**Comment by mokaddem** [comment]
@JakubOnderka thanks for the early review! :)

File: app/Controller/AttributesController.php (hunk 1 above)
Comment (mokaddem) [reply app/Controller/AttributesController.php]: Indeed!

File: app/Controller/AttributesController.php (hunk 2 above)
Comment (mokaddem) [reply app/Controller/AttributesController.php]: Yeah it's a dumb cut and paste from the previous `editSelected`. Nice catch

**Pull request merged**